astroid/inference_tip.py

```diff
@@ -18,9 +19,9 @@
     TransformFn,
 )
 
-_cache: dict[
+_cache: OrderedDict[
```

Comment:
We are on `3.7+` right? `Dicts` are ordered by default?

Reply:
```python
>>> d = {}
>>> d.popitem(last=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: dict.popitem() takes no keyword arguments
```

Reply:
The one thing `OrderedDict` is still good at :-)

Reply:
TIL :smile: